src/main/java/com/minecolonies/core/entity/citizen/citizenhandlers/CitizenFoodHandler.java

```diff
@@ -88,16 +88,18 @@ public Item getLastEaten()
     @Override
     public int checkLastEaten(final Item item)
     {
+        int foundIndex = -1;
+
         int index = -1;
         for (final Item foodItem : lastEatenFoods)
         {
+            index++;
             if (foodItem == item)
             {
-                return index;
+                foundIndex = index;
             }
-            index++;
         }
```

Comment:
Wouldn't it be sufficient to just move the `index++` to the beginning but keep the early return?  This is now returning the last index found instead of the first.